### PR TITLE
Expose IP addresses as plain strings (fixes #7)

### LINKED ISSUE #7: Accessing a server's public IPv4/IPv6
I want to access the public IPv4 and IPv6 addresses of a server to deploy DNS records after the server was created.

However, I am having trouble accessing the IP addresses because they are returned in a list of maps. As far as I understand, I can't filter a list of maps with current stable versions of terraform. 
This would only be possible with the new HCL engine that is going to be introduced with terraform 0.12. Even with the new HCL engine, accessing the public IP addresses would be unnecessary complex.

Other providers, for example the openstack provider, [expose the first detected IPv4 and IPv6 in a plain string field](https://www.terraform.io/docs/providers/openstack/r/compute_instance_v2.html#access_ip_v4). 
Adding those fields to the cloudscale.ch provider would be trivial. There is already existing code (`findIPv4AddrByType` and `findIPv6AddrByType`) that is adding the first public IPv4 or IPv6 address to the connection info for the server resource.

Would the maintainers of this provider accept a pull request that exposes the first public IPv4 and IPv6 as a string field?

### PR DESCRIPTION
**Pull request by ferdinandhuebner**
The additional attributes enable easier access to a server's IP addresses, for example to deploy DNS records. See #7.

I wasn't entirely sure about the best way to write acceptance tests for this. I hope that my implementation in `cloudscale/resource_cloudscale_server_test.go` makes sense for a terraform provider.

### CODE REVIEW

File: cloudscale/resource_cloudscale_server.go [hunk 1]
```diff
@@ -113,6 +113,22 @@ func getServerSchema() map[string]*schema.Schema {
...
+			Computed: true,
+		},
+		"private_ipv4": {
+			Type:     schema.TypeString,
+			Computed: true,
+		},
+		"private_ipv6": {
+			Type:     schema.TypeString,
+			Computed: true,
+		},
```
Comment (davidhalter) [inline cloudscale/resource_cloudscale_server.go:127]: I feel like we should call it `public_ipv4_address` and so on. I feel like others like digitalocean and others are doing that as well (and IMO it's the correct terminology).

Also: We definitely don't need `private_ipv6_address`, it's not necessary, because it will never be set. The private network v4 address could be useful, but I'm not sure if it's clever to include it.

The question is then arises if we should use `ipv4_address` instead, which just returns the first address (public if possible, otherwise private). Since this is a helper attribute anyway, I feel like we don't need to return everything.

What do you think? Do you have a use case for the private IPv4 address and the public one?

**Review by davidhalter — COMMENTED** [review]
Thanks a lot for this, I feel like it will be useful.

File: cloudscale/resource_cloudscale_server.go (hunk 1 above)
Comment (ferdinandhuebner) [reply cloudscale/resource_cloudscale_server.go:127]: Hi Dave,

thanks for the feedback. I can reword it to `public_ipv4_address` and remove the private v6 address.

We're deploying DNS records for both the public and private addresses, that's why I would like to keep the reference to the private one. 

At the moment, we're using the DNS record for the private address when we deploy kubernetes clusters and configure DNS of the private address in the "hostname-override" field in kubelet. 

Please let me know if you think it is okay to keep the private address.

File: website/docs/r/server.html.markdown [hunk 2]
```diff
@@ -58,6 +58,10 @@ In addition to the arguments listed above, the following computed attributes are
     * `device_path` - The path (string) to the volume on your server (e.g. `/dev/vda`)
     * `size_gb` - The size (int) of the volume in GB. Typically matches `volume_size_gb` or `bulk_volume_size_gb`.
     * `type` - A string. Either `ssd` or `bulk`.
+* `public_ipv4` - The first `public` IPv4 address of this server. The returned IP address may be `""` if the server does not have a public IPv4.
+* `private_ipv4` - The first `private` IPv4 address of this server. The returned IP address may be `""` if the server does not have private networking enabled.
+* `public_ipv6` - The first `public` IPv6 address of this server. The returned IP address may be `""` if the server does not have a public IPv6.
+* `private_ipv6` - The first `private` IPv6 address of this server. The returned IP address may be `""` if the server does not have private networking enabled.
```
Comment (davidhalter) [inline website/docs/r/server.html.markdown]: Please remove this mention. It's not correct anymore, because you removed it in the other files.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,11 @@
 ## 1.0.2 (Unreleased)
+
+
+IMPROVEMENTS:
+
+* Expose the first public/private IPv4 and IPv6 addresses as string attributes `public_ipv4`,
+  `public_ipv6`, `private_ipv4` and `private_ipv6`
```
Comment (davidhalter) [inline CHANGELOG.md]: Please remove `private_ipv6`.

File: cloudscale/resource_cloudscale_server.go (hunk 1 above)
Comment (davidhalter) [reply cloudscale/resource_cloudscale_server.go:127]: Hi Ferdinand. I'm now 100% fine with the changes, except that you forgot two things. Please fix those and we're good to merge.

**Review by davidhalter — APPROVED** [review]

**Comment by davidhalter** [comment]
It looks like I'm allowed to merge after all. I asked in their slack and I finally got an answer.
